spec/project_spec.rb

```diff
@@ -3689,4 +3823,120 @@ def callback(_args); end
+        variables: {'boolean_variable' => false},
+        variation_key: '177778'
+      )
+      expect(decisions['integer_single_variable_feature'].as_json).to eq(
+        enabled: true,
+        flag_key: 'integer_single_variable_feature',
+        reasons: [],
+        rule_key: 'test_experiment_integer_feature',
+        user_context: {attributes: {}, user_id: 'user1'},
+        variables: {'integer_variable' => 42},
+        variation_key: 'control'
+      )
+    end
+  end
+
+  describe '#decide_for_keys' do
+    it 'should get all the decisions for keys' do
+      keys = %w[
+        boolean_single_variable_feature
+        integer_single_variable_feature
+        boolean_feature
+        empty_feature
+      ]
+      stub_request(:post, impression_log_url)
+      user_context = project_instance.create_user_context('user1')
+      decisions = project_instance.decide_for_keys(user_context, keys)
+      expect(decisions.length).to eq(4)
+      expect(decisions['boolean_single_variable_feature'].as_json).to eq(
+        enabled: true,
+        flag_key: 'boolean_single_variable_feature',
+        reasons: [],
+        rule_key: '177776',
+        user_context: {attributes: {}, user_id: 'user1'},
+        variables: {'boolean_variable' => false},
+        variation_key: '177778'
+      )
+      expect(decisions['integer_single_variable_feature'].as_json).to eq(
+        enabled: true,
+        flag_key: 'integer_single_variable_feature',
+        reasons: [],
+        rule_key: 'test_experiment_integer_feature',
+        user_context: {attributes: {}, user_id: 'user1'},
+        variables: {'integer_variable' => 42},
+        variation_key: 'control'
+      )
+    end
+
+    it 'should get only enabled decisions for keys when ENABLED_FLAGS_ONLY is true' do
+      keys = %w[
+        boolean_single_variable_feature
+        integer_single_variable_feature
+        boolean_feature
+        empty_feature
+      ]
+      stub_request(:post, impression_log_url)
+      user_context = project_instance.create_user_context('user1')
+      decisions = project_instance.decide_for_keys(user_context, keys, [Optimizely::Decide::OptimizelyDecideOption::ENABLED_FLAGS_ONLY])
+      expect(decisions.length).to eq(2)
+      expect(decisions['boolean_single_variable_feature'].as_json).to eq(
+        enabled: true,
+        flag_key: 'boolean_single_variable_feature',
+        reasons: [],
+        rule_key: '177776',
+        user_context: {attributes: {}, user_id: 'user1'},
+        variables: {'boolean_variable' => false},
+        variation_key: '177778'
+      )
+      expect(decisions['integer_single_variable_feature'].as_json).to eq(
+        enabled: true,
+        flag_key: 'integer_single_variable_feature',
+        reasons: [],
+        rule_key: 'test_experiment_integer_feature',
+        user_context: {attributes: {}, user_id: 'user1'},
+        variables: {'integer_variable' => 42},
+        variation_key: 'control'
+      )
+    end
+  end
```

Comment:
I do not see tests for options
- INCLUDE_REASONS
- IGNORE_USER_PROFILE_SERVICE

Reply:
I also do not see any tests for "default_decide_options"

Reply:
IGNORE_USER_PROFILE_SERVICE tests are inside decision_service

Reply:
added tests for `INCLUDE_REASONS` too

Reply:
They look good. One thing I'd like to add is testing with IGNORE_USER_PROFILE_SERVICE, which skips "saving" as well as "reading" UPS.

Reply:
added assertions for both save and lookup in the existing tests